lib/sig0.js

```diff
@@ -81,22 +83,32 @@ sig0.makeKey = function makeKey(algorithm, priv) {
   return sig0.createKey(algorithm, pub);
 };
 
-sig0.createKey = function createKey(algorithm, publicKey) {
+sig0.createKey = function createKey(algorithm, publicKey, prefix) {
   assert((algorithm & 0xff) === algorithm);
   assert(Buffer.isBuffer(publicKey));
 
   const rr = new Record();
   const rd = new KEYRecord();
 
   rr.name = '.';
-  rr.class = classes.ANY;
+  rr.class = classes.IN;
   rr.type = types.KEY;
   rr.ttl = 0;
   rr.data = rd;
 
   rd.flags = 0;
   rd.protocol = 0;
   rd.algorithm = algorithm;
+
+  // https://tools.ietf.org/html/rfc4034#appendix-A.1.1
+  // PRIVATEDNS publicKey should be wire encoded and prefixed with algo name
+  if (algorithm === sig0.algs.PRIVATEDNS) {
```

Comment:
Should we have an assertion on `prefix` being a `string`, potentially along with its max size?

Reply:
I used `encoding.toName` because it has all the checks required for a domain name.